tests/uds.rs

```diff
@@ -130,6 +130,84 @@ fn is_send_and_sync() {
     assert_sync::<UnixDatagram>();
 }
 
+#[test]
+fn stream_pair() {
+    let (mut poll, mut events) = init_with_poll();
+    let (mut s1, mut s2) = assert_ok!(UnixStream::pair());
+
+    assert_ok!(poll
+        .registry()
+        .register(&s1, Token(0), Interests::READABLE | Interests::WRITABLE));
+    assert_ok!(poll
+        .registry()
+        .register(&s2, Token(1), Interests::READABLE | Interests::WRITABLE));
+
+    expect_events(
+        &mut poll,
+        &mut events,
+        vec![ExpectEvent::new(Token(0), Interests::WRITABLE)],
+    );
+
+    let mut buf = [0; DEFAULT_BUF_SIZE];
+    assert_would_block(s1.read(&mut buf));
+    let wrote = assert_ok!(s1.write(&DATA1));
+    assert_eq!(wrote, DATA1_LEN);
+    assert_ok!(s1.flush());
+
+    let read = assert_ok!(s2.read(&mut buf));
+    assert_would_block(s2.read(&mut buf));
+    assert_eq!(read, DATA1_LEN);
+    assert_eq!(&buf[..read], DATA1);
+    assert_eq!(read, wrote, "unequal reads and writes");
+
+    let wrote = assert_ok!(s2.write(&DATA2));
+    assert_eq!(wrote, DATA2_LEN);
+    assert_ok!(s2.flush());
+
+    let read = assert_ok!(s1.read(&mut buf));
+    assert_eq!(read, DATA2_LEN);
+    assert_eq!(&buf[..read], DATA2);
+    assert_eq!(read, wrote, "unequal reads and writes");
+}
+
+#[test]
+fn datagram_pair() {
+    let (mut poll, mut events) = init_with_poll();
+    let (s1, s2) = assert_ok!(UnixDatagram::pair());
+
+    assert_ok!(poll
+        .registry()
+        .register(&s1, Token(0), Interests::READABLE | Interests::WRITABLE));
+    assert_ok!(poll
+        .registry()
+        .register(&s2, Token(1), Interests::READABLE | Interests::WRITABLE));
+
+    expect_events(
+        &mut poll,
+        &mut events,
+        vec![ExpectEvent::new(Token(0), Interests::WRITABLE)],
+    );
+
+    let mut buf = [0; DEFAULT_BUF_SIZE];
+    assert_would_block(s1.recv(&mut buf));
+    let wrote = assert_ok!(s1.send(&DATA1));
+    assert_eq!(wrote, DATA1_LEN);
+
```

Comment:
Same as for `stream_pair`, needs needs a check for a readable event, same for next recv.